scripts/run_e2e_tests.py

```diff
@@ -697,14 +702,19 @@ def run_tests(args=None):
                                     # test is retried.
                                     pass # pragma: no cover
                                 return 'flake'
-    sys.exit(p.returncode)
+    if failure_seen:
+        cleanup_portserver(portserver_process)
+        cleanup()
+        return 'fail'
+    return 'pass'
 
 
 def main(args=None):
     """Run tests, rerunning at most MAX_RETRY_COUNT times if they flake."""
-    for _ in python_utils.RANGE(MAX_RETRY_COUNT):
+    for count in python_utils.RANGE(MAX_RETRY_COUNT):
+        python_utils.PRINT('\n\n###### RUN COUNT %d ######\n\n' % (count + 1))
         flake_state = run_tests(args=args)
-        if flake_state != 'flake':
+        if flake_state == 'pass':
             break
```

Comment:
Do we need to set the return code to ensure that if we fail 3 times in a row the CI check fails?

Reply:
Do you mean do a sys.exit(1) or something in main()?

Reply:
Yeah. Maybe a `sys.exit(0)` under `if flake_state == 'pass'` and then `sys.exit(1)` at the end of `main()`?

Reply:
sys.exit(0) will prevent the script from executing further, I think. I will add an exit with 1 in main. Thanks.

Reply:
Done.

Reply:
Yes `sysexit(0)` will stop script execution, but that's fine under `if flake_state == 'pass'` because we want to end execution anyway. I think your solution is fine though. One note: if `MAX_RETRY_COUNT` is 0, `flake_state` won't be defined yet. I don't think this is really a problem since `MAX_RETRY_COUNT` should always be at least 1, but we could make that assumption clear with an assertion.